influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -50,16 +53,19 @@ def get_func_n_trajectories(
     top_n = (
         avg_rewards.groupby(["env_name", "initial_state_id"])
         .apply(
-            lambda x: x.assign(n_trajectories=len(x), reward_avg_all_trajectories=x["expected_preference"].mean()).pipe(
```

Comment:
Maybe the previous line should also have `trajectory_id` as one of the things that is grouped by?